kokoro/build_binaries.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+# NOTE: This is WIP. For now it only builds a single binary. Once we are
+# certain this approach works, we will update it to build all binaries.
+
+set -o errexit
+set -o nounset
+set -o xtrace
+set -o pipefail
+
+export CI=kokoro
+
+export RUST_BACKTRACE=1
+export RUST_LOG=debug
+export XDG_RUNTIME_DIR=/var/run
+
+readonly SCRIPTS_DIR="$(dirname "$0")"
+# shellcheck source=scripts/common
+source "$SCRIPTS_DIR/common"
+
+readonly COMMIT_HASH="$(git rev-parse HEAD)"
+
+# Download the builder tool
+curl --location https://github.com/slsa-framework/slsa-github-generator/releases/download/v1.7.0/slsa-builder-docker-linux-amd64
+chmod +x slsa-builder-docker-linux-amd64
+
+# Build the binary. Should be made parametric.
+./slsa-builder-docker-linux-amd64 build \
+ --build-config-path buildconfigs/oak_restricted_kernel_simple_io_bin.toml \
+ --builder-image "$DOCKER_IMAGE_REPO_DIGEST" \
+ --git-commit-digest sha1:"$COMMIT_HASH" \
+ --source-repo git+https://github.com/project-oak/oak@refs/heads/main \
+ --subjects-path subjects.json \
+ --output-folder /tmp/build-outputs \
+ --verbose
```

Comment:
If we add all the binaries here, it will take a long time because each invocation will download all the deps from scratch (which is the proper thing to do when generating provenance to make sure builds don't interfere with each other).

I was thinking we could build the binaries together (i.e. in the same docker container, or even in the same nix shell) one after the other, for kokoro (instead of using the actual builder). In practice they should still be identical to the same binaries built separately (though we haven't checked).

But let's see if and when the latency of this job actually becomes an issue, and we can evaluate.